ui/src/app/archived-workflows/components/archived-workflow-list/archived-workflow-list.tsx

```diff
@@ -181,8 +181,54 @@ export class ArchivedWorkflowList extends BasePage<RouteComponentProps<any>, Sta
         return params;
     }
 
+    private filterStateObject(state: State, whitelistedStates: string[], blacklistedStates: string[]) {
+        const filteredState: State = {} as State;
```

Comment:
Just curious: why do we need both a blacklist and a whitelist? Is there a use case for that?

It seems if you pass `{whitelistedStates:['pagination'], blacklistedStates:[]}`, that's equivalent to  `{whitelistedStates:[], blacklistedStates:[]}`, so I'm wondering if `whitelistedStates` is really needed? 

I see that if you pass `{whitelistedStates:['pagination'], blacklistedStates:[`pagination`]}` then that would not be equivalent to `{whitelistedStates:[], blacklistedStates:[`pagination`]}` but I'm not sure if anyone would ever need to call it that way ?

Do let me know if I'm missing something, though. Is there some industry standard that uses both whitelists and blacklists together like this?

Reply:
Makes sense, I've removed whitelistedStates as we don't currently need it rn - I was initially thinking of a more generic function that filters the states (with either whitelist to select few states, or blacklist to reject few states)